include/Mat.h

```diff
@@ -55,14 +55,38 @@ template <class Type>
 class Const_MatIter;
 template <class Type, class Type2>
 class iMat;
+template <class Type = double>
+class Mat;
+
+struct DimInfo{
+    long ndim;
+    size_t dims[32];
+    size_t strides[32];
+    DimInfo(long a, size_t* b, size_t* c) : ndim(a){
```

Comment:
better input names. common formats:
```
dimsIn
_dims
```
or name them the same as locals and use `this->` on the instance ones